File: OOP/C# OOP/OOP Principles Part I/Library02/Student.cs
using System;

namespace Library02
{
    public class Student : Human
    {
        private float grade;

        //grade property
        public float Grade 
        {
            get
            {
                return this.grade;
            }
            set
            {
                this.grade = value;
            }
        }

        public Student(string firstName, string lastName, float grade)
        {
            this.FirstName = firstName;
            this.LastName = lastName;
            this.Grade = grade;
        }
    }
}

File: JavaScript/JS Applications/Advanced OOP/01,02.OOP/student.js
function Student(firstName, lastName, age, grade) {
	this.firstName = firstName;
	this.lastName = lastName;
	this.age = age;
	this.grade = grade;
	
	Person.apply(this, arguments);
	
	var baseIntroduce = this.introduce;
	
	this.introduce = function () {
		return baseIntroduce() + " Grade: " + grade;
	};
}

Student.prototype = new Person();
Student.prototype.constructor = Student;